pyvista/plotting/lookup_table.py

```diff
@@ -0,0 +1,754 @@
+        >>> lut.hue_range = (0.0, 0.33)
+        >>> lut.ramp = 's-curve'
+        >>> lut.plot()
+
+        Plot the linear ramp.
+
+        >>> import pyvista as pv
+        >>> lut = pv.LookupTable()
+        >>> lut.hue_range = (0.0, 0.33)
+        >>> lut.ramp = 'linear'
+        >>> lut.plot()
+
+        Plot the ``"sqrt"`` ramp.
+
+        >>> import pyvista as pv
+        >>> lut = pv.LookupTable()
+        >>> lut.hue_range = (0.0, 0.33)
+        >>> lut.ramp = 'sqrt'
+        >>> lut.plot()
+
+        """
+        return RAMP_MAP[self.GetRamp()]
+
+    @ramp.setter
+    def ramp(self, value: str):
+        try:
+            self.SetRamp(RAMP_MAP_INV[value])
+        except KeyError:
+            raise ValueError(f'`ramp` must be one of the following:\n{list(RAMP_MAP_INV.keys())}')
+        self._rebuild()
+
+    @property
+    def above_range_color(self) -> Optional[Color]:
+        """Return or set the above range color.
+
+        Any values above :attr:`LookupTable.table_range` will be rendered with this
+        color.
+
+        Examples
+        --------
+        Enable the usage of the above range color.
+
+        >>> import pyvista as pv
+        >>> lut = pv.LookupTable()
+        >>> lut.above_range_color = 'blue'
+        >>> lut.plot()
+
+        Disable the usage of the above range color.
+
+        >>> import pyvista as pv
+        >>> lut = pv.LookupTable()
+        >>> lut.above_range_color = None
+        >>> lut.plot()
+
+        """
+        if self.GetUseAboveRangeColor():
+            return Color(self.GetAboveRangeColor())
+        return None
+
+    @above_range_color.setter
+    def above_range_color(self, value):
+        if value is None:
+            self.SetUseAboveRangeColor(False)
+        else:
+            self.SetAboveRangeColor(*Color(value).float_rgba)
+            self.SetUseAboveRangeColor(True)
+
+    @property
+    def below_range_color(self) -> Optional[Color]:
+        """Return or set the below range color.
+
+        Any values below :attr:`LookupTable.table_range` will be rendered with this
+        color.
+
+        Examples
+        --------
+        Enable the usage of the below range color.
+
+        >>> import pyvista as pv
+        >>> lut = pv.LookupTable()
+        >>> lut.below_range_color = 'blue'
+        >>> lut.plot()
+
+        Disable the usage of the below range color.
+
+        >>> import pyvista as pv
+        >>> lut = pv.LookupTable()
+        >>> lut.below_range_color = None
+        >>> lut.plot()
+
+        """
+        if self.GetUseBelowRangeColor():
+            return Color(self.GetBelowRangeColor())
+        return None
+
+    @below_range_color.setter
+    def below_range_color(self, value):
+        if value is None:
+            self.SetUseBelowRangeColor(False)
+        else:
+            self.SetBelowRangeColor(*Color(value).float_rgba)
+            self.SetUseBelowRangeColor(True)
+
+    def _apply_cmap(self, cmap, n_values=256, flip=False):
```

Comment:
+1 for this not being private. LUT are an advanced feature and if a user is dabbling into them, I'd like to give them full control